cpp/action.h

```diff
@@ -0,0 +1,39 @@
+
+#pragma once
+
+#include <nlohmann/json.hpp>
+#include "emscripten_util.h"
+#include "point.h"
+#include "state.h"
+
+enum class ActionType { Unknown, TouchStart, TouchMove };
+
+JSON_SERIALIZE_ENUM(ActionType,
+                    {
+                        {ActionType::Unknown, "unknown"},
+                        {ActionType::TouchStart, "touchstart"},
+                        {ActionType::TouchMove, "touchmove"},
+                    });
+
+class Action {
+ public:
+  Action(ActionType type, Point point) : _type(type), _point(point) {}
+  Action() = default;
+  ;
+  // TODO(tdresser) - make this virtual, and use polymorphism?
```

Comment:
Having thought more, this is probably a bad idea.

Reply:
I dno't feel strongly, but I think I agree.